octoprint_mrbeam/static/js/material_settings.js

```diff
@@ -347,37 +396,87 @@ $(function () {
                     "Repeat the engraving once ore twice for a better result."
                 ),
                 safety_notes: "",
+                type: "product",
+                url: gettext("https://www.mr-beam.org/en/products/stempelgummi")
             },
             "Swiss stone pine": {
                 name: gettext("Swiss stone pine"),
                 img: "Zirbenholz.jpg",
                 description: "",
                 hints: "",
                 safety_notes: "",
+                type: null,
+                url: null
             },
             "Wood Sticker": {
                 name: gettext("Wood Sticker"),
                 img: "Wood-Sticker.jpg",
                 description: "",
                 hints: "",
                 safety_notes: "",
+                type: "product",
+                url: gettext("https://www.mr-beam.org/en/products/edelholzsticker")
             },
             "Solid Wood": {
                 name: gettext("Solid Wood"),
                 img: "Solid-Wood.jpg",
                 description: "",
                 hints: "",
                 safety_notes: "",
+                type: "collection",
+                url: gettext("https://www.mr-beam.org/en/collections/holz")
             },
             "Stainless Steel": {
                 name: gettext("Stainless Steel"),
                 img: "Stainless-Steel.jpg",
                 description: "",
                 hints: gettext("Engravings on stainless steel only work with our \"mark solid\" spray. Use line distance 0,1mm for the best result."),
                 safety_notes: gettext("Do not engrave stainless steel without the application of laser marking color."),
+                type: null,
+                url: null
             },
         };
         ///// EDIT MATERIAL SETTINGS ABOVE THIS LINE ////////`
+
+        self.constructShopifyURL = function (materialName, materialColor) {
+            // Validate material name and URL
+            if(self.validateMaterialName(materialName)  && self.validateMaterialUrl(materialName)) {
+                return self.addUrlReferral(self.constructMaterialURL(materialName, materialColor));
+            }
+            return self.addUrlReferral(self.MATERIAL_SHOPIFY_LINK);
+        }
+
+        self.validateMaterialName = function (materialName){
+            return materialName in self.materialSettingsDatabase;
```

Comment:
Not worth checking for None or string here?

Reply:
no